js/src/components/paid-ads/asset-group/asset-group-section.test.js

```diff
@@ -0,0 +1,66 @@
+jest.mock( '.~/components/adaptive-form', () => ( {
+	useAdaptiveFormContext: jest
+		.fn()
+		.mockName( 'useAdaptiveFormContext' )
+		.mockImplementation( () => {
+			return {
+				adapter: {
+					baseAssetGroup: { final_url: 'https://example.com' },
+					hasImportedAssets: false,
+					isEmptyAssetEntityGroup: false,
+					resetAssetGroup: jest.fn(),
+				},
+			};
+		} ),
+} ) );
+
+/**
+ * External dependencies
+ */
+import '@testing-library/jest-dom';
+import { render } from '@testing-library/react';
+
+/**
+ * Internal dependencies
+ */
+import AssetGroupSection from '.~/components/paid-ads/asset-group/asset-group-section';
+import { useAdaptiveFormContext } from '.~/components/adaptive-form';
+
+jest.mock( '.~/components/paid-ads/asset-group/asset-group-card', () =>
+	jest.fn( ( props ) => <div { ...props } /> ).mockName( 'AssetGroupCard' )
+);
+
+describe( 'AssetGroupSection', () => {
+	test( 'Component renders', () => {
+		const { queryByText } = render( <AssetGroupSection /> );
+		expect( queryByText( /Add additional assets/i ) ).toBeInTheDocument();
```

Comment:
Assertions checking for existence would be a bit better to use `getBy...` in terms of maintainability because it will throw a helpful error message if no element is found.

#### `queryBy...`

It only shows it received a `null`:

![image](https://github.com/woocommerce/google-listings-and-ads/assets/17420811/32a7c433-a895-4f43-9722-b520e3d2fc20)

#### `getBy...`

It prints out the whole document so the dev can see what's rendered and maybe why the query failed to find what they were looking for:

![image](https://github.com/woocommerce/google-listings-and-ads/assets/17420811/3a4bf820-fc4c-46d3-8d72-6e1be2a6ad3b)

Reply:
> Assertions checking for existence would be a bit better to use getBy...

I disagree with this one. Using `getByText` will throw an exception before calling `.toBeInTheDocument()`. So for example the test "Component not showing Tip if there are no imported assets" will break. 

Besides that, the error provided in the current implementation "Value must be an HTML Element or SVG element but NULL received", seems to me more concise and clearer in terms of checking existence.

Reply:
It's a suggestion got from the author of `@testing-library`: https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-query-variants-for-anything-except-checking-for-non-existence

> I disagree with this one. Using `getByText` will throw an exception before calling `.toBeInTheDocument()`. So for example the test "Component not showing Tip if there are no imported assets" will break.

I would say the break of the test doesn't matter because the test eventually failed regardless of which one.

> Besides that, the error provided in the current implementation "Value must be an HTML Element or SVG element but NULL received", seems to me more concise and clearer in terms of checking existence.

Proceeding to call `.toBeInTheDocument()` and showing it got a `null` doesn't actually help debugging any further. Often it's still necessary to add `screen.debug()` temporarily to print out the rendered DOM tree to find the cause, and this is exactly the thrown result of `getBy...` will print out.